client/java/src/main/java/com/linecorp/centraldogma/client/AbstractWatcher.java

```diff
@@ -95,6 +112,15 @@ private enum State {
         this.maxDelayMillis = maxDelayMillis;
         this.multiplier = multiplier;
         this.jitterRate = jitterRate;
+        this.meterRegistry = meterRegistry;
+        tags = Tags.of(
+                "project", projectName,
+                "repository", repositoryName,
+                "pathPattern", pathPattern,
+                // There is a possibility of using the same watcher for the same project, repo, and pathPattern.
+                // Therefore, the watcher’s hash code should be included as a tag.
+                "watcher_hash", String.valueOf(System.identityHashCode(this))
```

Comment:
Optional) What do you think of assigning an ID instead of a hashcode on the off-chance there is a conflict? It can also help users check how many watchers there are in a JVM easily.

e.g.

```
private static final AtomicLong WATCHER_ID = new AtomicLong();
...
"watcher_id", String.valueOf(WATCHER_ID.incrementAndGet())
```

Reply:
Ah, that sounds good too. If others prefer this approach, I’ll consider changing it.

Reply:
How about allowing users to set the name of a `Watcher`? If it is absent, we may create a name automatically.